plugin/ensime.vim

```diff
@@ -12,6 +12,7 @@ augroup END
 
 command! -nargs=* -range EnNoTeardown call ensime#com_en_no_teardown([<f-args>], '')
 command! -nargs=* -range EnTypeCheck call ensime#com_en_type_check([<f-args>], '')
+command! -nargs=* -range EnBufferedTypeCheck call ensime#com_en_buf_type_check([<f-args>], '')
```

Comment:
I agree @ches, adding this as the default style of typechecking w/ an option to disable if people want it is a nice simplicity win. I'll clean this up, remove the command & make it the default approach & then bang away on it for a while.